src/org/thoughtcrime/securesms/ConversationActivity.java

```diff
@@ -1121,7 +1121,7 @@ protected Void doInBackground(Object... param) {
 
           // for WEBXDC, drafts are just sent out as is.
           // for preparations and other cases, cleanup draft soon.
-          if (msg.getType() != DcMsg.DC_MSG_WEBXDC) {
+          if (msg == null || msg.getType() != DcMsg.DC_MSG_WEBXDC) {
```

Comment:
from the logic and the flow, if is maybe easier to understand if we move `if (msg.getType() != DcMsg.DC_MSG_WEBXDC) { ...` (including comment) three three lines down to the `if(msg!=null)` branch

Reply:
I don't see where `setDraft()` is called twice in the current code (and therefore think that if we don't call it here anymore, it won't be called at all):

There is a call `dcContext.setDraft(dcChat.getId(), msg);`, but it's in the `else` branch of `if (action==ACTION_SEND_OUT)`, so it's not called in the code flow we're talking about.

Reply:
you're right, i lost track of the brackets :)